marimo/_plugins/ui/_impl/table.py

```diff
@@ -346,3 +360,87 @@ def search(self, args: SearchTableArgs) -> Union[JSONType, str]:
         # Save the manager to be used for selection
         self._filtered_manager = result
         return result.limit(TableManager.DEFAULT_ROW_LIMIT).to_data()
+
+    def _apply_formatting(self, data: TableData) -> TableData:
+        # Apply formatting to the given data based on the format mapping.
+        import pandas as pd
+        import polars as pl
+        import pyarrow as pa
```

Comment:
the user may not have all these installed so this would break otherwise. we can move the logic for each dataframe time into their respective `TableManager`